awx/main/tasks.py

```diff
@@ -1917,6 +1928,60 @@ def build_env(self, inventory_update, **kwargs):
         env['AWX_PRIVATE_DATA_DIR'] = kwargs.get('private_data_dir', '')
         return env
 
+    def get_inventory_plugin_name(self, inventory_update):
+        if inventory_update.source in Inventory.SOURCE_PLUGINS:
+            return Inventory.SOURCE_PLUGINS[inventory_update.source]
+        if inventory_update.source == 'custom':
+            # TODO: allow user to specify which plugin to use
+            return None  # intentionally allow other defaults, like ini
+        return 'script'
+
+    def build_inventory(self, inventory_update, **kwargs):
+        src = inventory_update.source
+        if src in Inventory.SOURCE_PLUGINS and Version(kwargs['ansible_version']) >= Version('2.4.0'):
```

Comment:
This is a major sore point here. From discussions with the Ansible core team, we don't expect that all inventory scripts will be converted to plugins before the 2.5 release. Because of that, I think that the `Inventory.SOURCE_PLUGINS` dictionary will need to contain data on which version of Ansible _that particular plugin_ was introduced in. I think I remember @nitzmahone making this point.

Since some older versions of Ansible may not have some particular plugin, we'll have to continue to keep the inventory scripts until each ages out of the versions of Ansible that we support.